clippy_lints/src/types.rs

```diff
@@ -1646,6 +1652,25 @@ impl<'tcx> LateLintPass<'tcx> for Casts {
     }
 }
 
+fn show_unnecessary_cast<Num: Display>(
+    cx: &LateContext<'_>,
+    expr: &Expr<'_>,
+    num: Num,
```

Comment:
If the linted literal uses separators (e.g. `1_024 as u32`), they are lost in the suggestion. We should keep them and add a test about this case.

Reply:
Done